Insert value if entire range is selected

pavelKozlov opened the pull request:
When fixedDecimalScale is set to true and there is a value in the component, I am unable to overwrite the component's value entirely.

For example:
Initial component value: 4.50
Highlight/select the entire number
Type in a new number: 7

Result: Number does not change
Expected: 7.00

The description is taken from https://github.com/s-yadav/react-number-format/issues/142.
It also fixes https://github.com/s-yadav/react-number-format/issues/134.

s-yadav commented on src/number_format.js:
```diff
@@ -547,8 +547,10 @@ class NumberFormat extends React.Component {
     const diff = deletedIndex !== -1 ? lastValueParts[1].substring(0, deletedIndex) : '';
     const end = start + diff.length;
 
+    const isDeleteAll = lastValueParts[1].length === deletedIndex;
```
Wouldn't it will mark isDeleteAll for the case where all text after caretPos is selected. but there is text before caret position

s-yadav commented:
Can you check as other text are failing

pavelKozlov commented:
Hi @s-yadav,

You're right that the check if entire input value is replaced is not 100% accurate.
So I changed it.
I have also tested with different formats and masks from example sections and it works ok for me.
Maybe you can find other cases.
I have also added a test to cover new case and run all existent tests to be sure my changes don't break current code.

s-yadav commented:
Fixed this on 3.1.4. Let me know if this still exist
